crates/rune-macros/src/context.rs

```diff
@@ -111,6 +115,60 @@ pub(crate) struct FieldProtocol {
     custom: Option<syn::Path>,
 }
 
+pub(crate) struct TypeProtocol {
+    protocol: syn::Ident,
+    handler: Option<syn::Path>,
+}
+
+impl TypeProtocol {
+    pub fn expand(&self) -> TokenStream {
+        if let Some(handler) = &self.handler {
+            let protocol = &self.protocol;
+            return quote_spanned! {protocol.span()=>
+                module.associated_function(rune::runtime::Protocol::#protocol, #handler)?;
+            };
+        }
+        match self.protocol.to_string().as_str() {
+            "ADD" => quote_spanned! {self.protocol.span()=>
```

Comment:
`quote_spanned!` isn't very helpful when it's associated with large code blocks, therefore you should prefer just using `quote!` unless we've specifically added a UI test indicating there's a noted improvement in diagnostics.

Reply:
will add a ui test

Reply:
FTR, don't sweat it if you can't find a difference right now, you can just prefer to use `quote!`.